lib/inertia_rails/lazy.rb

```diff
@@ -0,0 +1,23 @@
+class InertiaRails::Lazy
+  def initialize(value = nil, &block)
+    @value = value
+    @block = block
+  end
+
+  def call
```

Comment:
call isn't actually used in the library, other than in `prop.respond_to?(:call)` in the renderer.

i decided to implement it anyway and add some explicit test coverage, since the rendering specs don't actually test it